js/bitstamp.js

```diff
@@ -1140,36 +1140,53 @@ module.exports = class bitstamp extends Exchange {
         return this.parseTradingFees (response);
     }
 
-    parseTransactionFees (balance) {
-        const withdraw = {};
-        const ids = Object.keys (balance);
-        for (let i = 0; i < ids.length; i++) {
-            const id = ids[i];
-            if (id.indexOf ('_withdrawal_fee') >= 0) {
-                const currencyId = id.split ('_')[0];
-                const code = this.safeCurrencyCode (currencyId);
-                withdraw[code] = this.safeNumber (balance, id);
-            }
-        }
-        return {
-            'info': balance,
-            'withdraw': withdraw,
-            'deposit': {},
-        };
-    }
-
     async fetchTransactionFees (codes = undefined, params = {}) {
         /**
          * @method
          * @name bitstamp#fetchTransactionFees
          * @description fetch transaction fees
+         * @see https://www.bitstamp.net/api/#balance
          * @param {[string]|undefined} codes not used by bitstamp fetchTransactionFees ()
          * @param {object} params extra parameters specific to the bitstamp api endpoint
          * @returns {[object]} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
          */
         await this.loadMarkets ();
-        const balance = await this.privatePostBalance (params);
-        return this.parseTransactionFees (balance);
+        const response = await this.privatePostBalance (params);
+        //
+        //  {
+        //     yfi_available: '0.00000000',
+        //     yfi_balance: '0.00000000',
+        //     yfi_reserved: '0.00000000',
+        //     yfi_withdrawal_fee: '0.00070000',
+        //     yfieur_fee: '0.000',
+        //     yfiusd_fee: '0.000',
+        //     zrx_available: '0.00000000',
+        //     zrx_balance: '0.00000000',
+        //     zrx_reserved: '0.00000000',
+        //     zrx_withdrawal_fee: '12.00000000',
+        //     zrxeur_fee: '0.000',
+        //     zrxusd_fee: '0.000',
+        //     ...
+        //  }
+        //
+        const result = {};
+        const ids = Object.keys (response);
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            if (id.indexOf ('_withdrawal_fee') >= 0) {
+                const currencyId = id.split ('_')[0];
+                const code = this.safeCurrencyCode (currencyId);
+                if (codes !== undefined && !this.inArray (code, codes)) {
+                    continue;
+                }
+                result[code] = {
+                    'deposit': undefined,
+                    'withdraw': this.safeNumber (response, id),
+                    'info': this.safeNumber (response, id),
```

Comment:
@AlgoQ We're losing a lot of the original info here, what if, in this  `info` we added all related fields? 

For instance:
```
code[ZRX] = {
     (...),
      'info': {
            zrx_available: '0.00000000',
            zrx_balance: '0.00000000',
            zrx_reserved: '0.00000000',
            zrx_withdrawal_fee: '12.00000000',
      }
}
```
wdyt?

Reply:
Only `zrx_available`, `zrx_balance`, `zrx_reserved` & `zrx_withdrawal_fee`?

Or also the other ones?
```
yfi_available: '0.00000000',
yfi_balance: '0.00000000',
yfi_reserved: '0.00000000',
yfi_withdrawal_fee: '0.00070000',
yfieur_fee: '0.000',
yfiusd_fee: '0.000',
```

Reply:
@AlgoQ all of them that was just an example 🙂